### ISSUE TITLE
HistogramObserver: forward returns detached tensor

### ISSUE THREAD

**Issue by IlichevSergey** [pytorch/pytorch#33545]
forward is implemented as follows:

    def forward(self, x_orig):
        # type: (Tensor) -> Tensor
        x = x_orig.detach()
        ...
        return x

## Expected behavior

I suppose that there is should be 

    def forward(self, x_orig):
        # type: (Tensor) -> Tensor
        x = x_orig.detach()
        ...
        return x_orig

## Environment

PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows 10 Enterprise
GCC version: (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0
CMake version: version 3.15.5

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.1.105
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 418.96
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] numpy==1.18.1
[pip] pytorch-memlab==0.0.2
[pip] torch==1.4.0
[pip] torchvision==0.5.0
[conda] blas 1.0 mkl
[conda] libmklml 2019.0.3 0
[conda] mkl 2019.5 281 anaconda
[conda] mkl-service 2.3.0 py36hb782905_0 anaconda
[conda] mkl_fft 1.0.12 py36h14836fe_0
[conda] mkl_random 1.0.2 py36h343c172_0
[conda] pytorch 1.4.0 py3.6_cuda101_cudnn7_0 pytorch
[conda] pytorch-memlab 0.0.2 pypi_0 pypi
[conda] torchvision 0.5.0 py36_cu101 pytorch

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a

**Comment by jerryzh168** [comment]
@raghuramank100 could you take a look?

**Comment by jerryzh168** [comment]
also cc @hx89

**Comment by dzhulgakov** [comment]
Yes, it's a bug - all other observers return `x_orig` and it's needed to make QAT work.
